Src/FluentAssertions/Equivalency/ConstraintCollectionEquivalencyStep.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Data;
+using System.Linq;
+
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Equivalency
+{
+    public class ConstraintCollectionEquivalencyStep : IEquivalencyStep
+    {
+        public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAssertionOptions config)
+        {
+            return typeof(ConstraintCollection).IsAssignableFrom(config.GetExpectationType(context));
+        }
+
+        public bool Handle(IEquivalencyValidationContext context, IEquivalencyValidator parent, IEquivalencyAssertionOptions config)
+        {
+            var subject = (ConstraintCollection)context.Subject;
+            var expectation = (ConstraintCollection)context.Expectation;
+
+            var subjectConstraints = subject.Cast<Constraint>().ToDictionary(constraint => constraint.ConstraintName);
+            var expectationConstraints = expectation.Cast<Constraint>().ToDictionary(constraint => constraint.ConstraintName);
+
+            var constraintNames = subjectConstraints.Keys.Union(expectationConstraints.Keys);
+
+            foreach (var constraintName in constraintNames)
+            {
+                AssertionScope.Current
+                    .ForCondition(subjectConstraints.TryGetValue(constraintName, out var subjectConstraint))
+                    .FailWith($"Expected constraint named {constraintName}, but did not find one");
```

Comment:
🤔 Don't you need to include the path inside the graph?

Reply:
I saw another implementation using string interpolation and latched onto it, but I'll change them all and maybe throw a `{reason}` in where it seems to make sense :-)